tsfile/src/main/java/org/apache/iotdb/tsfile/utils/BytesUtils.java

```diff
@@ -495,19 +518,35 @@ public static boolean bytesToBool(byte[] b, int offset) {
    * @param width bit-width
    */
   public static void longToBytes(long srcNum, byte[] result, int pos, int width) {
-    int temp = 0;
-    for (int i = 0; i < width; i++) {
-      temp = (pos + width - 1 - i) / 8;
-      try {
-        result[temp] = setByteN(result[temp], pos + width - 1 - i, getLongN(srcNum, i));
-      } catch (Exception e) {
-        LOG.error(
-            "tsfile-common BytesUtils: cannot convert a long {} to a byte array, pos {}, width {}",
-            srcNum,
-            pos,
-            width,
-            e);
+    if (width < 64) {
+      srcNum = srcNum & ~(-1L << width);
+    }
+    int cnt = pos % 8;
+    int index = pos / 8;
+    try {
+      while (width > 0) {
+        int m = width + cnt >= 8 ? 8 - cnt : width;
+        width -= m;
+        int mask = 1 << (8 - cnt);
+        cnt += m;
+        byte y = (byte) (srcNum >> width);
+        y = (byte) (y << (8 - cnt));
+        mask = ~(mask - (1 << (8 - cnt)));
+        result[index] = (byte) (result[index] & mask | y);
```

Comment:
Maybe `& mask` is unnecessary here because the bits between index `pos%8` - `pos%8+m` should always be `0`

Reply:
Thanks. But the original bits in `result` are not always 0 because the buffer is reused without clearing in DeltaEncoder.